server/files/models.py

```diff
@@ -24,3 +26,63 @@ class Meta:
         verbose_name = "File"
         verbose_name_plural = "Files"
         ordering = ("id",)
+
+
+class FileSource(models.Model):
+    """
+    Represents a source for files (an external URL)
+    """
+
+    DAILY = timedelta(days=1)
+    WEEKLY = timedelta(weeks=1)
+    UPDATE_INTERVALS = ((DAILY, "daily"), (WEEKLY, "weekly"))
+
+    notebook = models.ForeignKey(Notebook, on_delete=models.CASCADE)
+    # FIXME: add a validator for filename (for minimum length and maybe
+    # other things)
+    filename = models.CharField(max_length=MAX_FILENAME_LENGTH)
+    url = models.URLField()
+    update_interval = models.DurationField(null=True, choices=UPDATE_INTERVALS)
+
+    def __str__(self):  # pragma: no cover
+        return self.url
+
+    class Meta:
+        unique_together = ("notebook", "filename")
+        verbose_name = "File Source"
+        verbose_name_plural = "File Sources"
+        ordering = ("id",)
+        db_table = "file_source"
+
+
+class FileUpdateOperation(models.Model):
+    """
+    Represents a pending, ongoing, or completed operation to update a file
+    """
+
+    PENDING = 0
+    RUNNING = 1
+    COMPLETED = 2
+    FAILED = 3
+    OPERATION_STATUSES = (
+        (PENDING, "pending"),
+        (RUNNING, "running"),
+        (COMPLETED, "completed"),
+        (FAILED, "failed"),
+    )
+
+    file_source = models.ForeignKey(FileSource, on_delete=models.CASCADE)
+    scheduled = models.DateTimeField(auto_now_add=True)
```

Comment:
It's a good practice to name the datetime fields `scheduled_at`, `started_at` and `ended_at` to indicate that type they are and are not confused with boolean fields.

Reply:
that makes sense, and granted, this was me just spelunking through the code, so no pressing need for an actual change here

Reply:
I'll fix it, easier to do this now than later.